src/oom/oomd-util.h

```diff
@@ -76,7 +76,12 @@ static inline int compare_pgscan_and_memory_usage(OomdCGroupContext * const *c1,
         if (r != 0)
                 return r;
 
-        r = CMP((*c2)->pgscan, (*c1)->pgscan);
+        /* compare possible underflows first */
+        r = CMP((*c2)->pgscan > (*c2)->last_pgscan, (*c1)->pgscan > (*c1)->last_pgscan);
+        if (r != 0)
+                return r;
+
+        r = CMP((*c2)->pgscan - (*c2)->last_pgscan, (*c1)->pgscan - (*c1)->last_pgscan);
```

Comment:
Hm, `UINT64_MAX -> 1000` is ordered after `0 -> 1`. Is it expected behavior?

Reply:
Can you elaborate? I'm trying to sort the differences from large to small so I think the first `CMP` will sort the one with subtraction underflow at the end, and this second `CMP` will sort the subtracted difference as expected.

Reply:
When,
```
c2->pgscan = 1000;
c2->last_pgscan = UINT64_MAX;
c1->pgscan = 1;
c1->last_pgscan = 0;
```
then, `c1` is ordered earlier, no? If it is true, is it expected?

Reply:
Oh yup this is expected since essentially c1 increased pgscan but c2 decreased.

Reply:
I see, thanks. Then, one more question. What should we do when
```
c2->pgscan = 500;
c2->last_pgscan = 1000;
c1->pgscan = 200;
c1->last_pgscan = 1000;
```
i.e., pgscan of both contexts decrease. Is it not necessary to handle like the following?
```
r = CMP((*c2)->pgscan > (*c2)->last_pgscan, (*c1)->pgscan > (*c1)->last_pgscan);
if (r != 0)
        return r;

if ((*c2)->pgscan > (*c2)->last_pgscan)
        r = CMP((*c2)->pgscan - (*c2)->last_pgscan, (*c1)->pgscan - (*c1)->last_pgscan);
else
        r = CMP((*c1)->last_pgscan - (*c1)->pgscan, (*c2)->last_pgscan - (*c1)->pgscan);
if (r != 0)
        return r;
```

Reply:
I would think something like the following:
```
uint64_t pgscan1, pgscan2;

if ((*c1)->pgscan <= (*c1)->last_pgscan)
  pgscan1 = (*c1)->pgscan - (*c1)->last_pgscan;
else
  pgscan1 = UINT64_MAX - (*c1)->last_pgscan + (*c1)->pgscan;

if ((*c2)->pgscan <= (*c2)->last_pgscan)
  pgscan2 = (*c2)->pgscan - (*c2)->last_pgscan;
else
  pgscan2 = UINT64_MAX - (*c2)->last_pgscan + (*c2)->pgscan;

r = CMP(pgscan1, pgscan2);
if (r != 0)
        return r;
```

The fact that an overflow happened does not really tell us anything about the number of pgscan tables.

Reply:
yeah, the right approach with overflows is to have code paths that handle them properly, not just check they exist, i.e. please make sure that for the different overflow conditions there's a codepath that compares values properly